[Observability Onboarding] Merge feature branch to main

justinkambic opened the pull request:
## Summary

This PR tracks the overall progress on the updated Observability Onboarding Initiative's progress. ~This branch contains any commits that are considered completed but not ready to merge to Kibana `main`.~

~This patch introduces a feature flag and a scaffold for a new onboarding flow in the `observability_onboarding` plugin.~

Merging this will introduce the new experimental onboarding flow to `main`. This UI is not going to be the default at the time it is merged, it will be available at an alternative, non-discoverable URL.

For more information you can reference the [project board](https://github.com/orgs/elastic/projects/1502) for the initiative.

justinkambic committed:
Initial scaffolding for experimental onboarding plugin available.

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `/oblt-deploy-serverless` : Deploy a serverless Kibana instance using the Observability test environments.
- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)

</p>
</details>

flash1293 committed:
Merge branch 'main' into feat/obs-onboarding-flow

flash1293 committed:
Merge branch 'main' into feat/obs-onboarding-flow

justinkambic committed:
[Observability Onboarding] Only run new flow in Serverless (#179012)

## Summary

Previously we would run the experimental onboarding flow as long as the
A/B flag was enabled, regardless of env. We only want it to run in
Serverless, which is what this patch seeks to achieve.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Joe Reuter <johannes.reuter@elastic.co>

justinkambic committed:
Merge branch 'main' into feat/obs-onboarding-flow

justinkambic committed:
Merge branch 'main' into feat/obs-onboarding-flow

flash1293 committed:
Merge branch 'main' into feat/obs-onboarding-flow

flash1293 committed:
Merge branch 'main' into feat/obs-onboarding-flow

thomheymann committed:
Implement basic page layout for new landing page (#179327)

Resolves #178943

## :notebook: Summary

The observability onboarding page renders the new onboarding page:


![download](https://github.com/elastic/kibana/assets/190132/41f96946-8266-4ce9-9b38-6d2628a4fce0)

## :heavy_check_mark: Acceptance criteria
- Basic layout is rendered (header, questions, footer)
- User can choose between "Collect and analyze my logs", "Monitor my
applications performance", "Monitor my infrastructure"
- On hover, additional description is shown.

---------

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

justinkambic committed:
Merge branch 'feat/obs-onboarding-flow' of github.com:elastic/kibana into feat/obs-onboarding-flow

justinkambic committed:
Merge branch 'main' into feat/obs-onboarding-flow

justinkambic committed:
Merge branch 'main' into feat/obs-onboarding-flow

thomheymann committed:
[Observability Onboarding] Removes hover effect from use case selection (#180303)

## Summary

Removes hover effect from use case selection as per latest designs

## Screenshot

<img width="1602" alt="Screenshot 2024-04-08 at 15 53 49"
src="https://github.com/elastic/kibana/assets/190132/4de2d2b5-cb43-4836-b19a-a4eee306da9d">

justinkambic committed:
Merge branch 'main' into feat/obs-onboarding-flow

justinkambic committed:
Merge branch 'feat/obs-onboarding-flow' of github.com:elastic/kibana into feat/obs-onboarding-flow

justinkambic committed:
Merge branch 'main' into feat/obs-onboarding-flow

thomheymann committed:
[Observability Onboarding] Embed existing logs flows (#180054)

Resolves #178948

## Summary

Embeds existing custom and system log flows into new onboarding page. 

Note: As discussed there are several usability issues with this approach
which would need to be addressed before this can be merged into main.
This PR acts as more of a starting point rather than a finished feature.

## Screenshots

<img width="1372" alt="Screenshot 2024-04-04 at 17 10 46"
src="https://github.com/elastic/kibana/assets/190132/21281d93-ee61-437e-9ef5-249c49c29e75">

## Testing

The experimental onboarding flow can now be accessed using the hidden
URL: `/app/experimental-onboarding`

---------

Co-authored-by: Joe Reuter <johannes.reuter@elastic.co>

justinkambic committed:
Merge branch 'main' into feat/obs-onboarding-flow

elasticmachine commented:
Pinging @elastic/obs-ux-logs-team (Team:obs-ux-logs)

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

flash1293 reviewed (approved):
Approving for Logs UX

justinkambic committed:
Merge branch 'main' into feat/obs-onboarding-flow

flash1293 commented:
Tested again and confirmed that the new landing page is only shown when navigating to the `app/experimental-onboarding` app that's not linked to from anywhere. Merging to have a clean state to work on.
